sway/input/cursor.c

```diff
@@ -253,11 +253,32 @@ int cursor_get_timeout(struct sway_cursor *cursor) {
 	return timeout;
 }
 
+static enum sway_input_idle_source idle_source_from_device(
+		struct wlr_input_device *device) {
+	switch (device->type) {
+		case WLR_INPUT_DEVICE_KEYBOARD:
+			return IDLE_SOURCE_KEYBOARD;
+		case WLR_INPUT_DEVICE_POINTER:
+			return IDLE_SOURCE_POINTER;
+		case WLR_INPUT_DEVICE_TOUCH:
+			return IDLE_SOURCE_TOUCH;
+		case WLR_INPUT_DEVICE_TABLET_TOOL:
+			return IDLE_SOURCE_TABLET_TOOL;
+		case WLR_INPUT_DEVICE_TABLET_PAD:
+			return IDLE_SOURCE_TABLET_PAD;
+		case WLR_INPUT_DEVICE_SWITCH:
+			return IDLE_SOURCE_SWITCH;
+	}
+
+	assert(false);
```

Comment:
`abort()` is preferred because some release builds make assertions a no-op and that makes build fail (missing return value).